proxy/proxy.go

```diff
@@ -0,0 +1,66 @@
+// a client for interacting with the proxy - https://github.com/spec-tacles/proxy
+package proxy
+
+import (
+	"fmt"
+
+	"github.com/spec-tacles/go/broker"
+	"github.com/streadway/amqp"
+	"github.com/vmihailenco/msgpack"
+)
+
+type Proxy struct {
+	// The AMQP broker used to perform RPC calls to the proxy
+	Broker *broker.AMQP
+	// The event to call the requests on, usually `REQUEST`
+	RestEvent string
+	// The Discord bot token to perform requests with
+	// To perform Bearer requests, you'll have to specify the header yourself on each DoJSON or Do function
+	Token string
+}
+
+// generic, private request function
+func (p *Proxy) do(method string, path string, body *[]byte, options RequestOptions) (*Response, error) {
```

Comment:
This is still copying the options struct. Can we change this to accept a pointer to avoid unnecessarily copying the struct?